contractsSol5/KyberNetwork.sol

```diff
@@ -828,27 +828,27 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils4, IKyberNetwork, Reentra
         return _index;
     }
 
-    function calcTradeSrcAmount(uint srcDecimals, uint destDecimals, uint destAmount, uint[] memory rates,
-        uint[] memory currentSrcAmounts)
-        internal pure returns (uint srcAmount, uint[] memory newSrcAmounts)
+    function calcTradeSrcAmount(uint srcDecimals, uint destDecimals, uint destAmount, TradingReserves memory tradingReserves)
+        internal pure returns (uint srcAmount)
     {
-        uint totalDestAmount;
-        for(uint i = 0;i < currentSrcAmounts.length; i++) {
-            totalDestAmount += currentSrcAmounts[i] * rates[i];
+        uint totalWeightedDestAmount;
+        for (uint i = 0; i < tradingReserves.srcAmounts.length; i++) {
+            totalWeightedDestAmount += tradingReserves.srcAmounts[i] * tradingReserves.rates[i];
         }
 
         uint destAmountSoFar;
-        newSrcAmounts = new uint[](currentSrcAmounts.length);
-
-        for (uint i = 0; i < currentSrcAmounts.length; i++) {
-            uint destAmountSplit = i == (currentSrcAmounts.length - 1) ?
-                (destAmount - destAmountSoFar) : destAmount * currentSrcAmounts[i] * rates[i] / totalDestAmount;
+        for (uint i = 0; i < tradingReserves.srcAmounts.length; i++) {
+            uint currentSrcAmt = tradingReserves.srcAmounts[i];
+            uint destAmountSplit = i == (tradingReserves.srcAmounts.length - 1) ?
+                (destAmount - destAmountSoFar) :
+                destAmount * currentSrcAmt * tradingReserves.rates[i] / totalWeightedDestAmount;
             destAmountSoFar += destAmountSplit;
 
-            newSrcAmounts[i] = calcSrcQty(destAmountSplit, srcDecimals, destDecimals, rates[i]);
-            require(newSrcAmounts[i] <= currentSrcAmounts[i], "new src amount is high");
+            uint newSrcAmt = calcSrcQty(destAmountSplit, srcDecimals, destDecimals, tradingReserves.rates[i]);
```

Comment:
So this is for readability. Can save directly into tradingReserves.srcAmounts[i] actually